### PR TITLE
fix: revert in-repo prisma and fix client generation and schema migrations

### PR DESCRIPTION
**Pull request by Encord-davids**

### CODE REVIEW

File: src/encord_active/lib/db/prisma_init.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+prisma_run = silence_stdout(run)
+
+GENERATED_PRISMA_SCHEMA_FILE = Path(prisma_module_file).parent / "schema.prisma"
+PRISMA_SCHEMA_FILE = Path(__file__).parent / "schema.prisma"
+
+
+def ensure_prisma_db(db_path: Path):
+    env = {"MY_DATABASE_URL": f"file:{db_path}"}
+    failure = prisma_run(["migrate", "deploy", f"--schema={PRISMA_SCHEMA_FILE}"], env=env)
+    # TODO: remove me before release
```
Comment (eloy-encord) [inline src/encord_active/lib/db/prisma_init.py:18]: what about this todo? is ok to go with it to main?

File: src/encord_active/lib/db/connection.py [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
-from pathlib import Path
 from typing import Optional
 
-from encord_active.lib.db.__autogen__.cli.prisma import run
-from encord_active.lib.db.__autogen__.client import Prisma
-from encord_active.lib.db.__autogen__.types import DatasourceOverride
-from encord_active.lib.file_structure.base import BaseProjectFileStructure
+from encord_active.lib.db.prisma_init import generate_prisma_client
+
+try:
```
Comment (eloy-encord) [inline src/encord_active/lib/db/connection.py:6]: should we move the try/except to `prisma_init` module so after calling that module we can ensure that prisma effectively works?

File: src/encord_active/lib/project/project_file_structure.py [hunk 3]
```diff
@@ -2,7 +2,8 @@
 from pathlib import Path
 from typing import Iterator, NamedTuple, Optional, Union
 
-from encord_active.lib.db.__autogen__ import models
+from prisma import models
```
Comment (eloy-encord) [inline src/encord_active/lib/project/project_file_structure.py:5]: probably we should add prisma initialization before importing the models, or moving this import to after the prisma db connection import

File: src/encord_active/lib/file_structure/base.py [hunk 4]
```diff
@@ -2,7 +2,8 @@
 from pathlib import Path
 from typing import Iterator, Optional, Union
 
-from encord_active.lib.db.__autogen__ import models
+from prisma import models
```
Comment (eloy-encord) [inline src/encord_active/lib/file_structure/base.py:5]: probably we should add prisma initialization before importing the models

**Review by eloy-encord — CHANGES_REQUESTED** [review]

File: src/encord_active/lib/db/prisma_init.py (hunk 1 above)
Comment (Encord-davids) [reply src/encord_active/lib/db/prisma_init.py:18]: we want it to go to main for a short while so everyone who already used prisma can start all projects and reset them. after that we can remove

File: src/encord_active/lib/project/project_file_structure.py (hunk 3 above)
Comment (Encord-davids) [reply src/encord_active/lib/project/project_file_structure.py:5]: Good point, I'll move the try/except block to the `db` module init

**Review by frederik-encord — APPROVED** [review]

**Review by eloy-encord — APPROVED** [review]

**Pull request merged**